tests/generic/test_units.py

```diff
@@ -0,0 +1,76 @@
+# coding: utf-8
+# Copyright (c) Max-Planck-Institut für Eisenforschung GmbH - Computational Materials Design (CM) Department
+# Distributed under the terms of "New BSD License", see the LICENSE file.
+
+import numpy as np
+from pyiron_base.generic.units import PyironUnitRegistry, UnitConverter
+import unittest
+import pint
+
+pint_registry = pint.UnitRegistry()
+
+
+class TestUnits(unittest.TestCase):
+
+    def test_units(self):
+        base_units = PyironUnitRegistry()
+        base_units.add_quantity(quantity="energy", unit=pint_registry.eV)
+        self.assertRaises(ValueError, base_units.add_quantity, quantity="energy", unit="eV")
+        dim_less_q = pint_registry.Quantity
+        base_units.add_quantity(quantity="dimensionless_integer_quantity", unit=dim_less_q(1), data_type=int)
+        code_units = PyironUnitRegistry()
+        # Define unit kJ/mol
+        code_units.add_quantity(quantity="energy",
+                                unit=pint_registry.kilocal / (pint_registry.mol * pint_registry.N_A))
+        code_units.add_labels(labels=["energy_tot", "energy_pot"], quantity="energy")
+        # Raise Error for undefined quantity
```

Comment:
Comments like this could live inside `self.assertRaises(..., msg='Raise Error for undefined quantity') to make the test output a little bit more readable.